packages/sdk/src/models/item-map.ts

```diff
@@ -0,0 +1,34 @@
+import { MinimalSlot, SlotHelpers } from './local-slot';
+
+/**
+ * This class adds `toMap` which is a specialization of `toNameSlotMap` for the particular type of
+ * Slot that the Item contains.
+ */
+export default class ItemMap<SlotType extends MinimalSlot> {
+  /**
+   * Represent an Item as a map from Slot.names to Slots.
+   * @param item
+   */
+  static toNameSlotMap<S extends MinimalSlot>(slots: S[]): Record<string, S> {
+    function getSlotName(slot: S): string {
+      return slot.name || SlotHelpers.nameFromLabel(slot.label!);
+    }
+
+    const map: Record<string, S> = {};
+
+    slots.forEach(slot => {
+      map[getSlotName(slot)] = slot;
+    });
+
+    return map;
+  }
+
+  constructor(public readonly slots: SlotType[]) {}
+
+  /**
+   * The Item represented as a map from Slot names to Slots.
+   */
+  toMap(): Record<string, SlotType> {
```

Comment:
Good points! I was hesitant to call it `toJSON` since it doesn't actually serialize the item: so you would break the `JSON.parse(JSON.stringify(X)) === X` expectation. I also couldn't call it `toObject` either haha! Perhaps I could call it `toSlotMap` or `toAssignments` or ...?

Reply:
>so you would break the JSON.parse(JSON.stringify(X)) === X expectation

I don't think that's an expectation:

![image](https://user-images.githubusercontent.com/9100419/99472770-3ae8c300-2999-11eb-8b11-c08addcbad99.png)

![image](https://user-images.githubusercontent.com/9100419/99472845-5e137280-2999-11eb-9a7b-1c5463174647.png)


I think it's just much more convenient that serializing one of these instances comes out as something that isn't `[Object object]`

Reply:
Damn, the world is so much more well-behaved inside my head

Reply:
I'm still not 100% ok with calling it toJSON as it is so lossy. Which would you prefer:
- getNameSlotMap
- getSlotsByName
- keySlotsByName

Reply:
I'd prefer `.slots` or `.values`  as a plain getter.

Reply:
I like `values` a lot, as then `assignSlots` can be renamed as `set values`.

Reply:
I still liked it, so I did both: `ItemMap` now has `slotsByName` = (old `toMap`) and `get values` which is a name/value map representation of the slots. `ItemChange` has both `assignValues` (like `Object.assign`) and `set values` which satisfies (roughly) `item.values = X ==> item.values == X`.